src/cryptonote_protocol/cryptonote_protocol_handler.inl

```diff
@@ -361,6 +362,194 @@ namespace cryptonote
   }
   //------------------------------------------------------------------------------------------------------------------------
   template<class t_core>
+  int t_cryptonote_protocol_handler<t_core>::handle_notify_new_fluffy_block(int command, NOTIFY_NEW_FLUFFY_BLOCK::request& arg, cryptonote_connection_context& context)
+  {
+    LOG_PRINT_CCONTEXT_L0("NOTIFY_NEW_FLUFFY_BLOCK (hop " << arg.hop << ")");
+    if(context.m_state == cryptonote_connection_context::state_normal)
+    {
+      m_core.pause_mine();
+      
+      block new_block;
+      transaction miner_tx;
+      if(parse_and_validate_block_from_blob(arg.b.block, new_block))
+      {
+        std::list<blobdata> have_tx;
+        std::list<crypto::hash> need_tx_hashes;
+        
+        transaction tx;
+        crypto::hash tx_hash;
+        std::unordered_map<crypto::hash, blobdata> tx_hash_to_blob;        
+        BOOST_FOREACH(auto& tx_blob, arg.b.txs)
+        {
```

Comment:
Hmm, my main goal here was to get the transactions into the pool if they weren't already, so I definitely need to add a check to see if it's already in there and avoid doing all the extra work. 

Is there a better way to verify and add to pool without using the handle_incoming_tx? I'll have a look later.

Worth adding a check that the transactions included are part of the tx_hashes, otherwise drop peer?